Database/MongoDB/Query.hs

```diff
@@ -1527,7 +1851,12 @@ runCommand1 c = runCommand [c =: (1 :: Int)]
 
 eval :: (MonadIO m, Val v) => Javascript -> Action m v
 -- ^ Run code on server
-eval code = at "retval" <$> runCommand ["$eval" =: code]
+eval code = do
+    p <- asks mongoPipe
+    let sd = P.serverData p
+    if maxWireVersion sd <= 7
```

Comment:
7 or 17?

Reply:
7 is correct. `eval` has been removed in mongoDB 4.2.